### PR TITLE
[Issue 2667] Create Wk3 Seeker AI Curriculum

### LINKED ISSUE #2667: Create Wk3 Seeker AI Curriculum
### Description

Create topical / themed topic outlines that address 3 of 6 weeks of 75-minute AI sessions.

### Location

[AI folder](https://github.com/Techtonica/curriculum/tree/main/artificial-intelligence)

### Why Is This Important

Program alum are now facing the recurring question of "How do you use AI" in interviews as well in their day to day tasks when on a team. In efforts of continued learning, this issue seeks to provide some context for how to do some of the basic level access to using AI as a tool as a SWE.

### Acceptance Criteria

Topic: AI for Testing & Code Quality
- Prerequisite for this topic outline should include an assumption that the consumer has completed 1000 hours of full stack development
- Create 3 of 6 75-min AI session [topic outlines](https://github.com/Techtonica/curriculum/blob/main/_templates/topic-outline.md). 

In addition to usual topic outline sections, each write up should include: 
- specific tasks of activities that participants should complete as well as specific resources to use and how to turn items in
- how to show up and participate in 75-min AI sessions

### PR DESCRIPTION
**Pull request by daaimah123**
### 📝 Description
Problem: Program alum are now facing the recurring question of "How do you use AI" in interviews as well in their day to day tasks when on a team. In efforts of continued learning, this issue seeks to provide some context for how to do some of the basic level access to using AI as a tool as a SWE.

Task: Create topical / themed topic outlines that address 3 of 6 weeks of 75-minute AI sessions. 

Topic: AI for Debugging & Documentation

### 🔂 Changes Made
New `job-seeker-ai-series` folder added to [AI folder](https://github.com/Techtonica/curriculum/tree/main/artificial-intelligence)

### ⚙️ Related Issue
Issue Number: #2667

### 🎁 Acceptance Criteria
- Prerequisite include an assumption that the consumer has completed 1000 hours of full stack development
- 30 min independent practice
- 45 min guided practice
- specific tasks of activities that participants should complete as well as specific resources to use and how to turn items in
- how to show up and participate in each session

### CODE REVIEW

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+- **Test anatomy:** Setup, action, assertion—and how AI sometimes skips critical steps
+- **Parameterized tests:** Using AI to generate multiple test cases efficiently
+- **Mocking and fixtures:** Asking AI to generate mock data and test doubles
+- **Error case testing:** Ensuring AI generates tests for failure modes, not just happy paths
+- **Framework-specific syntax:** Getting AI to generate tests in your specific tool (Jest, pytest, etc.)
+
+### Code Quality & Coverage Analysis
+- **Coverage metrics:** Line coverage, branch coverage, function coverage—and what each means
+- **Coverage gaps:** Using AI to identify which code paths aren't tested
+- **Edge case thinking:** AI's common blind spots (null/undefined, empty collections, boundary values)
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: I've also found that AI sometimes struggles with dates / times

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 2]
```diff
@@ -0,0 +1,616 @@
...
+| **Prompting for Test Generation** | Guide (Reference Week 1 & 2 resources on prompting; apply to testing context)| Specific techniques for getting AI to generate useful tests | 
+| **ChatGPT for Unit Testing** | Blog/Video (YouTube: search "ChatGPT unit testing tutorial") | Practical walkthrough of AI-assisted test writing | 
+| **[Testing Edge Cases](https://www.cypress.io/blog/2023/11/testing-edge-cases/)** | Article | Systematic approach to identifying edge cases AI might miss |
+
+### Edge Cases & Boundary Testing
+
+| Resource | Type | Why It Matters | 
+|-----|-----|-----|
+| **[Boundary Value Analysis](https://www.softwaretestinghelp.com/boundary-value-analysis-test-case-design-technique/)** | Testing Technique | Systematic approach to identifying edge cases | 
+| **[Equivalence Partitioning](https://en.wikipedia.org/wiki/Equivalence_partitioning)** | Testing Technique | Group test cases into equivalence classes; ensure comprehensive coverage | [https://en.wikipedia.org/wiki/Equivalence_partitioning
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]:
```suggestion
| **[Equivalence Partitioning](https://en.wikipedia.org/wiki/Equivalence_partitioning)** | Testing Technique | Group test cases into equivalence classes; ensure comprehensive coverage | 
```

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 3]
```diff
@@ -0,0 +1,616 @@
...
+
+| Resource | Type | Why It Matters | Link
+|-----|-----|-----|-----
+| **"How Do You Approach Testing?"** | Interview Question | Common interview question; practice articulating your philosophy | Blind.com, interview prep blogs
+| **Test Coverage Discussion** | Conversation Starter | Discuss tradeoffs: speed vs. quality, coverage goals | GitHub discussions, engineering blogs
+| **Real-World Testing Scenarios** | Case Study | Study how teams balance AI-assisted testing with quality | Search company engineering blogs (Stripe, Vercel, GitHub, etc.)
+
+### Optional: Going Deeper
+
+| Resource | Type | For Those Interested
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]:
```suggestion
| Resource | Type | For Those Interested | Link
```

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 4]
```diff
@@ -0,0 +1,616 @@
...
+- Offer specific feedback using the structure above
+- Be open to suggestions; this is collaborative learning
+- Share one insight you gained from the peer review
+
+## Supplemental Materials
+
+### Free Testing Frameworks & Tools
+
+| Tool | Language | Free Tier | Best For
+|-----|-----|-----|-----
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]:
```suggestion
|-----|-----|-----|-----|
```

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 5]
```diff
@@ -0,0 +1,616 @@
+# Job Seeker AI Session Week 3: AI for Testing & Code Quality
+
+## Prerequisites
+- Completion of 1000+ hours of PERN full stack development experience
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]:
```suggestion
- Completion of 1000+ hours of PERN full-stack development experience
```

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 6]
```diff
@@ -0,0 +1,616 @@
...
+| **Test Coverage Discussion** | Conversation Starter | Discuss tradeoffs: speed vs. quality, coverage goals | GitHub discussions, engineering blogs
+| **Real-World Testing Scenarios** | Case Study | Study how teams balance AI-assisted testing with quality | Search company engineering blogs (Stripe, Vercel, GitHub, etc.)
+
+### Optional: Going Deeper
+
+| Resource | Type | For Those Interested
+|-----|-----|-----|
+| **Property-Based Testing** | Advanced | Generate test cases automatically; ensures robustness | [Python](https://hypothesis.readthedocs.io/) or [JavaScript](https://fast-check.dev/)
+| **Mutation Testing** | Advanced | Test the tests; ensure your assertions catch real bugs | [https://stryker-mutator.io/](https://stryker-mutator.io/)
+| **Contract Testing** | Advanced | Test API contracts between services; relevant for fullstack | [https://pact.foundation/](https://pact.foundation/)
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]:
```suggestion
| **Contract Testing** | Advanced | Test API contracts between services; relevant for full-stack | [https://pact.foundation/](https://pact.foundation/)
```

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 7]
```diff
@@ -0,0 +1,616 @@
...
+**The Three Levels of Test Evaluation:**
+1. **Does it run?** (Technical correctness)
+  - No syntax errors
+  - Correct imports/fixtures
+  - Assertions are valid
+2. **Does it test?** (Behavioral correctness)
+  - Assertion validates the expected behavior
+  - Input setup is realistic
+  - Test name matches what's being tested
+3. **Is it maintainable?** (Code quality)
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md:374]: Also important that the tests are in the correct file / directory, with the correct naming convention

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 8]
```diff
@@ -0,0 +1,616 @@
...
+| **[Using NYC for Coverage Reports](https://github.com/istanbuljs/nyc)** | Tutorial | Analyze JavaScript test coverage; identify gaps | 
+| **[Using Coverage.py for Python](https://coverage.readthedocs.io/)** | Tutorial | Generate coverage reports; understand coverage data | 
+
+### AI-Assisted Test Generation
+
+| Resource | Type | Why It Matters | 
+|-----|-----|-----|
+| **Prompting for Test Generation** | Guide (Reference Week 1 & 2 resources on prompting; apply to testing context)| Specific techniques for getting AI to generate useful tests | 
+| **ChatGPT for Unit Testing** | Blog/Video (YouTube: search "ChatGPT unit testing tutorial") | Practical walkthrough of AI-assisted test writing | 
+| **[Testing Edge Cases](https://www.cypress.io/blog/2023/11/testing-edge-cases/)** | Article | Systematic approach to identifying edge cases AI might miss |
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: I get a 404 on this link

https://www.cypress.io/blog/2023/11/testing-edge-cases/

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 9]
```diff
@@ -0,0 +1,616 @@
...
+
+### Edge Cases & Boundary Testing
+
+| Resource | Type | Why It Matters | 
+|-----|-----|-----|
+| **[Boundary Value Analysis](https://www.softwaretestinghelp.com/boundary-value-analysis-test-case-design-technique/)** | Testing Technique | Systematic approach to identifying edge cases |
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: I get a 404 on this link

https://www.softwaretestinghelp.com/boundary-value-analysis-test-case-design-technique/

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 10]
```diff
@@ -0,0 +1,616 @@
...
+| **[pytest Fixtures](https://docs.pytest.org/en/stable/fixture.html)** | Official Docs | Reusable test setup; prevents duplication | 
+| **[Factory Pattern in Tests](https://github.com/thoughtbot/factory_bot)** | Article | Generate realistic test data efficiently |  (Ruby example, but pattern applies everywhere)
+
+### Test Maintainability & Refactoring
+
+| Resource | Type | Why It Matters | 
+|-----|-----|-----|
+| **[Test Smell: A Quick Reference](https://testsmells.org/)** | Article | Recognize and fix common test code smells | 
+| **[DRY in Tests](https://en.wikipedia.org/wiki/Don't_repeat_yourself)** | Best Practices | Avoid test duplication; maintain tests long-term | 
+| **[Meaningful Test Names](https://www.typescriptlang.org/docs/handbook/testing.html) (reference section)** | Article | Write test names that document behavior |
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: 404 on this link

https://www.typescriptlang.org/docs/handbook/testing.html

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md [hunk 11]
```diff
@@ -0,0 +1,616 @@
...
+| **[Test Smell: A Quick Reference](https://testsmells.org/)** | Article | Recognize and fix common test code smells | 
+| **[DRY in Tests](https://en.wikipedia.org/wiki/Don't_repeat_yourself)** | Best Practices | Avoid test duplication; maintain tests long-term | 
+| **[Meaningful Test Names](https://www.typescriptlang.org/docs/handbook/testing.html) (reference section)** | Article | Write test names that document behavior | 
+
+### Security & Robustness Testing
+
+| Resource | Type | Why It Matters | 
+|-----|-----|-----|
+| **[OWASP Testing Guide](https://owasp.org/www-project-web-security-testing-guide/)** | Reference | Common vulnerabilities to test for | 
+| **[Security Test Cases](https://portswigger.net/web-security/testing)** | Article | What edge cases protect against injection, type coercion, etc. |
```
Comment (monikkaelyse) [inline artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: 404 on this link

https://portswigger.net/web-security/testing

**Review by monikkaelyse — CHANGES_REQUESTED** [review]
Really great work! Just some minor issues, most pressing is the broken links

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 1 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 2 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 3 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 4 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 5 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 7 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md:374]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 8 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 9 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 10 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 11 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in https://github.com/Techtonica/curriculum/commit/447e6ec9adb87974d8169ae07cfc6ac61f9a1980

File: artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md (hunk 6 above)
Comment (daaimah123) [reply artificial-intelligence/job-seeker-ai-series/job-seeker-ai-session3.md]: Addressed in d96c76d

**Review by monikkaelyse — APPROVED** [review]
LGTM!
